pandas/tests/extension/test_arrow.py

```diff
@@ -3143,3 +3144,15 @@ def test_to_numpy_temporal(pa_type):
     expected = np.array(expected, dtype=object)
     assert result[0].unit == expected[0].unit
     tm.assert_numpy_array_equal(result, expected)
+
+
+def test_arrowextensiondtype_dataframe_repr():
+    # GH 54062
+    df = pd.DataFrame(
+        pd.period_range("2012", periods=3),
+        columns=["col"],
+        dtype=ArrowDtype(ArrowPeriodType("D")),
+    )
+    result = repr(df)
+    expected = "     col\n0  15340\n1  15341\n2  15342"
```

Comment:
where deos 15340 come from?  i'd expect something like "2012-01-01"

Reply:
I think this is because `ArrowPeriodType` stores period data as `pa.int64`. Since `pyarrow.ExtensionType` is only used for serialization it appears it doesn't go through period reprs

Reply:
im unclear on if this should be considered "wrong"?  or just orthogonal to what this PR is doing?  seems like this would be an unhelpful repr to give a user

Reply:
I think it's a orthogonal incorrect behavior. This PR addresses a bug where `pyarrow.ExtensionType`s didn't even show a `repr`

Reply:
make sense

Reply:
maybe a comment that this test shouldn't be interpreted as saying this repr is "correct"?

Reply:
Sure. Added a comment